Source/WTF/wtf/Assertions.h

```diff
@@ -754,6 +754,24 @@ constexpr bool assertionFailureDueToUnreachableCode = false;
 
 #endif /* ASSERT_ENABLED */
 
+/* CONJECTURE_ASSERT is only used to facilitate on-going analysis work to test conjectures
+   about the code. We want to be able to land these in the code base for some time to enable
+   extended testing.
+
+   If the conjecture is proven false it, the CONJECTURE_ASSERT should either be removed or
+   updated to test a new conjecture. If the conjecture is proven true, the CONJECTURE_ASSERT
+   should either be promoted to an ASSERT or RELEASE_ASSERT as appropriate, or removed if
+   deemed of low value.
+
+   The number of CONJECTURE_ASSERTs should not be growing unboundedly, and they should not
+   stay in the codebase perpetually.
```

Comment:
Maybe we should add a comment here that all `CONJECTURE_ASSERT` should have a comment with the date they were added / radar (maybe?) so it's easy to see how long they've been sitting around?

Reply:
I'd thought about this, and think that `git blame` provides sufficient info.  As long as the number of `CONJECTURE_ASSERT`s is not runaway large, this should be sufficient.  This is a tradeoff vs keeping the code readable.  If this proves to be not true in the future, we can change how this works e.g. making the 2nd argument a date for the introduction of the `CONJECTURE_ASSERT`.  For now, I'll stick with YAGNI.